homeassistant/components/ruckus_unleashed/config_flow.py

```diff
@@ -70,30 +75,36 @@ async def async_step_user(
                 errors["base"] = "cannot_connect"
             except InvalidAuth:
                 errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unknown error")
             else:
-                await self.async_set_unique_id(info[KEY_SYS_SERIAL])
-                self._abort_if_unique_id_configured()
-                return self.async_create_entry(
-                    title=info[KEY_SYS_TITLE], data=user_input
+                if self._reauth_entry is None:
+                    await self.async_set_unique_id(info[KEY_SYS_SERIAL])
```

Comment:
Theoretically possible to have multiple unleashed networks, on different subnets, and point at the wrong one.  
But apart from me, I doubt there's anyone doing this at home.

I did try changing the unique_id from the Master AP's serial # to the Unleashed ID (which is a unique key for the entire cluster), so I can tell it's the same Unleashed network as before. But I was asked to remove this change and implement later, rather than fixing it to include a migration.

I would like to return ConfigEntryAuthFailed when the I get a ConnectionError (i.e. nothing is responding at the IP), since the old Master AP might've been decommissioned, but a different member AP has taken over. Is this OK?  
  > And ideally I'd store the IPs of member APs into the config entry so I can try these before giving up and prompting. Is this allowed?

Reply:
I suggest we make sure that the unique_id is the same when re-authenticating a config entry.

ConfigEntryAuthFailed is not raised within the config flow. That's a question for a different PR, I think. It's ok to try connecting again before raising auth failed, yes.